packages/destination-actions/src/destinations/intercom/trackEvent/index.ts

```diff
@@ -0,0 +1,88 @@
+import type { ActionDefinition } from '@segment/actions-core'
+import dayjs from '../../../lib/dayjs'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+
+const action: ActionDefinition<Settings, Payload> = {
+  title: 'Track Event',
+  description: 'Submit a data event to Intercom.',
+  fields: {
+    event_name: {
+      type: 'string',
+      required: true,
+      description:
+        'The name of the event. Names are treated as case insensitive. Periods (.) and dollars ($) in event names are replaced with hyphens.',
+      label: 'Event Name',
+      default: {
+        '@path': '$.event'
+      }
+    },
+    created_at: {
+      type: 'datetime',
+      required: true,
+      description: 'A datetime in Unix timestamp format (seconds since Epoch).',
+      label: 'Created At',
+      default: {
+        '@path': '$.timestamp'
+      }
+    },
+    user_id: {
+      type: 'string',
+      description: "The user's ID; required if no email provided.",
+      label: 'User ID',
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    email: {
+      type: 'string',
+      description: "The user's email; required if no User ID provided.",
+      label: 'Email',
+      format: 'email',
+      default: {
+        '@if': {
+          exists: { '@path': '$.properties.email' },
+          then: { '@path': '$.properties.email' },
+          else: { '@path': '$.traits.email' }
```

Comment:
`track` events won't have a `traits` object.

Reply:
You could have the second field be `context.traits.email` since `track` calls can house `traits` within the `context` object.

```suggestion
          else: { '@path': '$.context.traits.email' }
```

Reply:
chatted with @NolanChan and we will leave the default as just `properties.traits`  (no coalesce)